File: 프로그래머스/프로그래머스_Lv3/최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        int[] array = new int[n];
        if(s/n == 0) {
            int[] answer = new int[1];
            answer[0] = -1;
            return answer;
        }

        for(int index = 0; index < n; index ++) {
            array[index] = s/n;
        }
        int left = s % n;
        int index = n - 1;
        while (left > 0) {
        array[index]++;
        index = (index - 1 + n) % n;
        left--;
        }
        return array;
    }
}

File: 프로그래머스/프로그래머스_Lv3/최고의 집합/최고의 집합.js
function solution(n, s) {
  const array = new Array(n).fill(Math.floor(s / n));
  if (array.reduce((acc, current) => acc + current, 0) === 0) return [-1];
  let left = s % n;
  let index = n - 1;
  while (left > 0) {
    array[index]++;
    index = (index - 1 + n) % n;
    left--;
  }
  return array;
}
